### PR TITLE
[continuous-deploy-fingerprint] Add platform as input

### PR DESCRIPTION
**Pull request by NoZiL**
Playing with the new action for sdk 51, I've noticed that the platform is missing.
This is an attempt to handle this case.

### CODE REVIEW

**Review by brentvatne — APPROVED** [review]
can you update the readme as well?

File: src/actions/continuous-deploy-fingerprint.ts [hunk 1]
```diff
@@ -30,63 +33,103 @@ export async function continuousDeployFingerprintAction(input = collectContinuou
...
+      platform = 'android';
+      break;
+    case 'ios':
+      platform = 'ios';
+      break;
+  }
 
-  info(`Android fingerprint: ${androidFingerprintHash}`);
-  info(`iOS fingerprint: ${iosFingerprintHash}`);
+  async function buildAndroid() {
```
Comment (wschurman) [inline src/actions/continuous-deploy-fingerprint.ts:46]:
```suggestion
  async function buildAndroidIfNecessaryAsync() {
```

File: src/actions/continuous-deploy-fingerprint.ts [hunk 2]
```diff
@@ -30,63 +33,103 @@ export async function continuousDeployFingerprintAction(input = collectContinuou
...
+    if (platform === 'ios') return;
```
Comment (wschurman) [inline src/actions/continuous-deploy-fingerprint.ts:47]: small nit (null is clearer in result type than void):
```suggestion
    if (platform === 'ios') {
      return null;
    }
```

File: src/actions/continuous-deploy-fingerprint.ts [hunk 3]
```diff
@@ -30,63 +33,103 @@ export async function continuousDeployFingerprintAction(input = collectContinuou
...
-    profile: input.profile,
-    fingerprintHash: iosFingerprintHash,
-    excludeExpiredBuilds: isInPullRequest,
-  });
-  if (iosBuildInfo) {
-    info(`Existing iOS build found with matching fingerprint: ${iosBuildInfo.id}`);
-  } else {
-    info(`No existing iOS build found for fingerprint, starting a new build...`);
-    iosBuildInfo = await createEASBuildAsync({ cwd: input.workingDirectory, platform: 'ios', profile: input.profile });
+  async function buildIos() {
```
Comment (wschurman) [inline src/actions/continuous-deploy-fingerprint.ts:79]:
```suggestion
  async function buildIosIfNecessaryAsync() {
```

File: src/actions/continuous-deploy-fingerprint.ts [hunk 4]
```diff
@@ -30,63 +33,103 @@ export async function continuousDeployFingerprintAction(input = collectContinuou
...
+    if (platform === 'android') return;
```
Comment (wschurman) [inline src/actions/continuous-deploy-fingerprint.ts:80]:
```suggestion
    if (platform === 'android') {
      return null;
    }
```

File: src/actions/continuous-deploy-fingerprint.ts [hunk 5]
```diff
@@ -30,63 +33,103 @@ export async function continuousDeployFingerprintAction(input = collectContinuou
...
+        platform: 'ios',
+        profile: input.profile,
+      });
+    }
+
+    return { fingerprint: iosFingerprintHash, buildInfo: iosBuildInfo };
   }
 
-  const builds = [androidBuildInfo, iosBuildInfo];
+  const [android, ios] = await Promise.all([buildAndroid(), buildIos()]);
```
Comment (wschurman) [inline src/actions/continuous-deploy-fingerprint.ts:112]:
```suggestion
  const [androidResult, iosResult] = await Promise.all([
    buildAndroidIfNecessaryAsync(), 
    buildIosIfNecessaryAsync(),
  ]);
```

File: src/actions/continuous-deploy-fingerprint.ts [hunk 6]
```diff
@@ -30,63 +33,103 @@ export async function continuousDeployFingerprintAction(input = collectContinuou
...
   if (!isInPullRequest) {
     info(`Skipped comment: action was not run from a pull request`);
   } else {
     const messageId = `continuous-deploy-fingerprint-projectId:${projectId}`;
-    const messageBody = createSummaryForUpdatesAndBuilds({ config, projectId, updates, builds, options: input });
+    const messageBody = createSummaryForUpdatesAndBuilds({
+      config,
+      projectId,
+      updates,
+      builds: [android?.buildInfo, ios?.buildInfo].filter(e => e !== undefined) as BuildInfo[],
```
Comment (wschurman) [inline src/actions/continuous-deploy-fingerprint.ts:130]:
```suggestion
      builds: [android?.buildInfo, ios?.buildInfo].filter(e => e !== null) as BuildInfo[],
```

**Review by wschurman — CHANGES_REQUESTED** [review]
Good addition! Just a few stylistic changes, then I'll merge!

**Comment by NoZiL** [comment]
Hey @brentvatne, I added the input in the readme, tell me if something else needs to be added.
Hey @wschurman, sorry for the delay, I've dealt with the conflicts and added your suggestions. I also added the strict null checks in tsconfig to lint accordingly.

**Review by wschurman — APPROVED** [review]

**Comment by wschurman** [comment]
Thanks for the contribution!

**Pull request merged**